File: SRC/CPCDOS EDITOR/Host/RandomUtil.cs
using System;
using System.Drawing;

namespace OSMaker.Host
{
    /// <summary>
    /// Generated random color. It is used by MyRootDesigner
    /// </summary>
    public class RandomUtil
    {
        internal const int MaxRGBInt = 255;
        private static Random rand = null;

        public RandomUtil()
        {
            if (rand is null)
                InitializeRandoms(new Random().Next());
        }

        private void InitializeRandoms(int seed)
        {
            rand = new Random(seed);
        }

        public virtual Color GetColor()
        {
            byte rval, gval, bval;
            rval = (byte)GetRange(0, MaxRGBInt);
            gval = (byte)GetRange(0, MaxRGBInt);
            bval = (byte)GetRange(0, MaxRGBInt);
            var c = Color.FromArgb(rval, gval, bval);
            return c;
        }

        public int GetRange(int nMin, int nMax)
        {
            // Swap max and min if min > max
            if (nMin > nMax)
            {
                int nTemp = nMin;
                nMin = nMax;
                nMax = nTemp;
            }

            if (nMax != int.MaxValue)
                System.Threading.Interlocked.Increment(ref nMax);
            int retVal = rand.Next(nMin, nMax);
            return retVal;
        }
    }
}

File: SRC/OSMHost/Host/RandomUtil.vb
Imports System
Imports System.Drawing

Namespace OSMaker
    ''' <summary>
    ''' Generated random color. It is used by MyRootDesigner
    ''' </summary>
    Public Class RandomUtil
        Friend Const MaxRGBInt = 255
        Private Shared rand As Random = Nothing

        Public Sub New()
            If rand Is Nothing Then InitializeRandoms(New Random().Next())
        End Sub

        Private Sub InitializeRandoms(ByVal seed As Integer)
            rand = New Random(seed)
        End Sub

        Public Overridable Function GetColor() As Color
            Dim rval, gval, bval As Byte
            rval = CByte(GetRange(0, MaxRGBInt))
            gval = CByte(GetRange(0, MaxRGBInt))
            bval = CByte(GetRange(0, MaxRGBInt))
            Dim c = Color.FromArgb(rval, gval, bval)
            Return c
        End Function

        Public Function GetRange(ByVal nMin As Integer, ByVal nMax As Integer) As Integer
            ' Swap max and min if min > max
            If nMin > nMax Then
                Dim nTemp = nMin
                nMin = nMax
                nMax = nTemp
            End If

            If nMax <> Integer.MaxValue Then Threading.Interlocked.Increment(nMax)
            Dim retVal = rand.Next(nMin, nMax)
            Return retVal
        End Function
    End Class
End Namespace
